libcxx/docs/DesignDocs/NodiscardExtensions.rst

```diff
@@ -0,0 +1,34 @@
+======================================
+``[[nodiscard]]`` extensions in libc++
+======================================
+
+Libc++ adds ``[[nodiscard]]`` to functions in a lot more places than the
```

Comment:
Generally speaking this wording needs to be rebased on top of the Tokyo discussion + policy paper.